include/openthread/instance.h

```diff
@@ -102,6 +102,22 @@ otInstance *otInstanceInit(void *aInstanceBuffer, size_t *aInstanceBufferSize);
  */
 otInstance *otInstanceInitSingle(void);
 
+/**
+ * This static method initializes the OpenThread instance.
+ *
+ * This function initializes OpenThread and prepares it for subsequent OpenThread API calls. This function must be
+ * called before any other calls to OpenThread. This method utilizes static buffer to initialize the OpenThread
+ * instance.
+ *
+ * This function is available and can only be used when support for multiple OpenThread static instances is enabled.
+ *
+ * @param[in] aIdx The index of the OpenThread instance to initialize.
+ *
+ * @returns  A pointer to the new OpenThread instance.
+ *
+ */
+otInstance *otInstanceInitMultiple(uint8_t aIdx);
```

Comment:
Should it be wrapped under `OPENTHREAD_CONFIG_MULTIPLE_STATIC_INSTANCE_ENABLE`?

Reply:
I don't think we #ifdef forward declarations in header files in OpenThread. OPENTHREAD_CONFIG_MULTIPLE_STATIC_INSTANCE_ENABLE may not be available where the include/openthread/* files are used. 
@abtink any opinion on this?

Reply:
@MarekPorwisz  you are correct. 

We cannot and should not use any `OPENTHREAD_CONFIG_{}` in any public or platform header files. These header may be included in other projects where these definitions are not accessible.

We usually document that certain API requires certain `OPENTRHEAD_CONFIG_`